src/tree/annotation/poseidon.rs

```diff
@@ -24,48 +24,6 @@ pub struct PoseidonAnnotation {
 }
 
 impl PoseidonAnnotation {
-    /// Create a new poseidon annotation from a generic node implementation
-    pub fn from_generic_node<C, A>(node: &C) -> Self
-    where
-        C::Leaf: PoseidonLeaf,
-        A: PoseidonTreeAnnotation<C::Leaf>,
-        A: Annotation<C::Leaf>,
-        C: Compound<A>,
-    {
```

Comment:
Why is this  removed and brought into `combine`?

Reply:
I would bring @krl into the discussion and see his oppinion.

Reply:
As below, removing unneccesary trait bounds is always a good thing.